pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -203,6 +203,10 @@ public void deleteSegmentFile() {
   private final String _resourceDataDir;
   private final IndexLoadingConfig _indexLoadingConfig;
   private final Schema _schema;
+  // Semaphore for each partitionId only. See the comments in {@link RealtimeTableDataManager}.
+  private final Semaphore _partitionConsumerSemaphore;
+  // A boolean flag to check whether the current thread acquires the semaphore, so that the semaphore be released only once within the same thread.
```

Comment:
```suggestion
  // A boolean flag to check whether the current thread has acquired the semaphore, so that the semaphore be released only once within the same thread.
```